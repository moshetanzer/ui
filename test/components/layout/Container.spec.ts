import { describe, it, expect } from 'vitest'
import { UContainer } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Container', () => {
  it.each([
    ['basic case', {}],
    ['with as', { props: { as: 'article' } }],
    ['with default slot', { slots: { default: () => 'Default slot' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UContainer.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UContainer)
    expect(html).toMatchSnapshot()
  })
})
