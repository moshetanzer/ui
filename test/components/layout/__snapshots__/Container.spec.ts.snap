// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > renders basic case correctly 1`] = `"<div class="mx-auto px-4 sm:px-6 lg:px-8 max-w-7xl"></div>"`;

exports[`Container > renders with as correctly 1`] = `"<article class="mx-auto px-4 sm:px-6 lg:px-8 max-w-7xl"></article>"`;

exports[`Container > renders with default slot correctly 1`] = `"<div class="mx-auto px-4 sm:px-6 lg:px-8 max-w-7xl">Default slot</div>"`;
